crates/sui-metrics-proxy/src/channels.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use axum::body::Bytes;
+use prometheus_parse::Scrape;
+use tokio::sync::mpsc::Receiver;
+use tracing::error;
+/// NodeMetric is a placeholder for metric data
+/// tbd in future PR
+#[derive(Debug)]
+pub struct NodeMetric {
+    pub host: String,
+    pub data: Bytes, // raw post data from node
+}
+
+/// An UpstreamConsumer accepts bytes from calling clients and is
+/// responsible for sending them on to upstream services to store
+/// the relayed metric
+pub struct UpstreamConsumer {
+    pub network: String,
+    pub receiver: Receiver<NodeMetric>,
+}
+
+impl UpstreamConsumer {
+    pub fn new(network: String, receiver: Receiver<NodeMetric>) -> Self {
+        Self { network, receiver }
+    }
+    pub async fn run(&mut self) {
+        while let Some(nm) = self.receiver.recv().await {
+            let network = self.network.to_owned();
+            tokio::spawn(async move {
+                let Ok(data) = std::str::from_utf8(&nm.data) else {
+                    error!("unable to decode bytes from relayed metrics");
+                    return;
+                };
+                let lines = data.lines().map(|s| Ok(s.to_owned()));
+                let mut metrics = Scrape::parse(lines).expect("sdf");
+                let Some(l) = metrics.samples.first_mut() else {
+                    error!("unable to get mutable handle to sample data");
+                    return;
+                };
+                l.labels.insert("host".to_string(), nm.host);
+                l.labels.insert("network".to_string(), network);
```

Comment:
Do we need to add these labels to all of the metrics? or does what you've done do that

Reply:
i backported some code from another local PR into here so you can see a bit of what's next. it's not complete yet, not w/o the chain data.

Reply:
> I need to convert it to protobufs to submit to mimir
 If that's the case then we can probably change the encoding that the clients send to already be in the protobuf format